engine/collection/message_hub/message_hub.go

```diff
@@ -244,27 +247,30 @@ func (h *MessageHub) processQueuedMessages(ctx context.Context) error {
 // processQueuedTimeout performs actual processing of model.TimeoutObject, as a result of successful invocation
 // broadcasts timeout object to consensus committee.
 // No errors are expected during normal operations.
-func (h *MessageHub) processQueuedTimeout(timeout *model.TimeoutObject) error {
-	log := timeout.LogContext(h.log).Logger()
-	log.Info().Msg("processing timeout broadcast request from hotstuff")
+func (h *MessageHub) processQueuedTimeout(timeout *messages.ClusterTimeoutObject) error {
+	logContext := h.log.With().
+		Uint64("timeout_newest_qc_view", timeout.NewestQC.View).
+		Uint64("timeout_tick", timeout.TimeoutTick).
+		Hex("timeout_newest_qc_block_id", timeout.NewestQC.BlockID[:]).
+		Uint64("timeout_view", timeout.View)
+
+	if timeout.LastViewTC != nil {
+		logContext.
+			Uint64("last_view_tc_view", timeout.LastViewTC.View).
+			Uint64("last_view_tc_newest_qc_view", timeout.LastViewTC.NewestQC.View)
+	}
+	log := logContext.Logger()
 
 	// Retrieve all collection nodes in our cluster (excluding myself).
-	recipients, err := h.state.Final().Identities(filter.And(
-		filter.In(h.cluster),
-		filter.Not(filter.HasNodeID(h.me.NodeID())),
-	))
+	recipients, err := h.state.Final().Identities(h.clusterIdentityFilter)
 	if err != nil {
 		return fmt.Errorf("could not get cluster members for broadcasting timeout: %w", err)
 	}
-	// create the timeout message
-	msg := &messages.ClusterTimeoutObject{
-		View:       timeout.View,
-		NewestQC:   timeout.NewestQC,
-		LastViewTC: timeout.LastViewTC,
-		SigData:    timeout.SigData,
-	}
 
-	err = h.con.Publish(msg, recipients.NodeIDs()...)
+	err = h.con.Publish(timeout, recipients.NodeIDs()...)
+	if errors.Is(err, network.EmptyTargetList) {
+		return nil
+	}
```

Comment:
I think this code duplicates behaviour two lines below: https://github.com/onflow/flow-go/blob/e96e75f2aca1837728a2305104e35ed2064268d1/engine/collection/message_hub/message_hub.go#L275-L278
wondering if I am missing something 🤔 

```suggestion
```